packages/shared/auth0-client/src/index.ts

```diff
@@ -16,6 +20,25 @@ export default class Auth0Client {
     this.clientSecret = clientSecret;
   }
 
+  async deleteUser(userId: number): Promise<APIResponse<boolean>> {
+    const httpClient = await this.getMachineToMachineInstance();
+
+    return httpClient.delete(`/users/auth0|p${userId}`, { validateStatus: responseCodeIs(204), })
+      .then(_ => {
+        return successResponse(true);
+      })
+      .catch((error: AxiosError) => {
+        if (error.response) {
+          switch (error.response.status) {
+            case 404:
+              return errorResponse(`User with id "auth0|p${userId}" not found`, ResponseStatus.NotFound, error);
```

Comment:
This error message exposes the inner working of Auth0 to the client, i.e. the `auth0|p` ID prefix. I've made a concious effort to avoid leaking this stuff outside of this class, so everywhere above here should just deal with the numerical user ID.

Where we report the similar error message elsewhere, we use:

```
'Auth0 user with ID [' + userId + '] not found'
```

where `userId` is the numerical ID of the user that aligns with Sierra.